File: c#/singleton/Ejm1/LoggerFactory.cs
using System.Collections.Generic;
using Libs;
namespace Singleton.Ejm1
{
    /// <summary>
    /// Fabrica de objetos
    /// </summary>
    class LoggerFactory
    {
        /// <summary>
        /// Determina si el parámetro de configuración FileLogging=ON
        /// </summary>
        /// <returns>True si está en ON, false si esta en OFF</returns>
        public bool IsFileLogginEnabled()
        {
            Dictionary<string, string> properties = Properties.GetProperties(@"d:\logger.properties");
            if (properties["FileLogging"].Equals("ON"))
                return true;
            else
                return false;
        }

        /// <summary>
        /// Fabrica de objetos
        /// </summary>
        /// <returns>Un objeto FileLogger ó un objeto ConsoleLogger, segun el criterio</returns>
        public Logger getLogger()
        {
            if (IsFileLogginEnabled())
                return FileLogger.getFileLogger();//Observar este cambio
            else
                return new ConsoleLogger();
        }
    }
}

File: java/singleton/ej2/LoggerFactory.java
package patrones.singleton.ej2;

import java.io.IOException;
import java.util.Properties;

public class LoggerFactory {
	public boolean isFileLoggingEnabled() {
		Properties p = new Properties();
		try {
			p.load(ClassLoader.getSystemResourceAsStream("logger.properties"));
			String fileLoggingValue = p.getProperty("FileLogging");
			if (fileLoggingValue.equalsIgnoreCase("ON") == true)
				return true;
			else
				return false;
		} catch (IOException e) {
			return false;
		}
	}

	//Factory Method
	public Logger getLogger() {
		if (isFileLoggingEnabled()) {
			//Notar este cambio
			return FileLogger.getFileLogger();
		} else {
			return new ConsoleLogger();
		}
	}
}
